test/lsp-mode-test.el

```diff
@@ -43,4 +43,28 @@
   (with-temp-buffer
     (test-tcp-client-enable)))
 
+(ert-deftest lsp-define-whitelist ()
+  (lsp-define-stdio-client test-stdio-client "test language"
+                           (lambda () "/tmp/baz/")
+                           '("/bin/false"))
+  (should (fboundp 'test-stdio-client-whitelist-add))
+  (should (fboundp 'test-stdio-client-whitelist-remove))
+  (should (not lsp-project-whitelist))
+  (with-temp-buffer
+    (test-stdio-client-whitelist-add)
+    (should (equal lsp-project-whitelist '("^/tmp/baz/$") ))
+    ;; Should not add a duplicate
+    (test-stdio-client-whitelist-add)
+    (should (equal lsp-project-whitelist '("^/tmp/baz/$") ))
+    ;; testing remove
+    (customize-save-variable 'lsp-project-whitelist
+           (add-to-list 'lsp-project-whitelist (concat "^" (regexp-quote "/tmp/foo") "$")))
```

Comment:
lsp--as-regex?

Reply:
I was fighting with it at one stage so made everything explicit. Will change it though.